# EDA on my favorite TV show Shark Tank India

## Introduction

<p>I did a deep dive into the data from Shark Tank India, my favorite TV show. Basically, I used SQL to sift through the data and find interesting stuff about the number of pitches, which ones got funding, where the brands were located, and details about the average equity and investments made by different sharks. The goal was to uncover patterns and connections in the data</p>

I web scraped wikipedia data of Shark tank season 1 and transformed it using Excel. I have uploaded the ready file in the github repo for this project. 

#### Dataset Link: https://github.com/Swapppyy/EDA-on-Shark-Tank-India/blob/main/data.csv

### The aim of this notebook is to answer below questions:

- <p>Total Number of Episodes

- <p>Total number of pitches

- <p>Pitches that received Funding

- <p>Total amount invested by Sharks

- <p>Distribution of genders on the teams
   

- <p>Highest investement and equity taken by any of the shark

- <p>Partner deals of sharks

- <p>Table of my 4 favorite sharks and their investments information

- <p>Startup in which the highest amount has been invested in each domain/sector

## Data Analysis using SQL

#### To showcase my work i am using a different approach using Jupyter Notebook

<p>As i am using my Jupyter Notebook i need to establish connection with SQL on my local machine.</p>
<p>Using a SQL Magic Extension: Jupyter Notebook has a built-in extension called "SQL Magic" that allows you to write SQL queries directly in a notebook cell.</p>

<p>You can use any SQL tool you like to run this queries after loading the dataset </p>

In [ ]:
%load_ext sql
%sql sqlite:///C:/Users/mahaj/Downloads/data.db


In [3]:
%sql select * from data

 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(1, 'BluePine Industries', 2.0, 1.0, 'Delhi', 'Frozen Momos', 'Food', '75 lakhs for 16% equity', 75, 50.0, 0, 0, 16.0, 5.0, '35-40', 3, 25.0, 5.333333333, 0.0, 0.0, 0.0, 0.0, 25.0, 5.33, 25.0, 5.33, None, None, None, None, 3.0, 'Ama-Ash-Vin'),
 (1, 'Booz scooters', 1.0, 0.0, 'Ahemdabad', 'Renting e-bike for mobility in private spaces', 'Automobile', '40 lakhs for 50% equity', 40, 40.0, 0, 0, 50.0, 15.0, '25-30', 1, 20.0, 25.0, 0.0, 0.0, 0.0, 0.0, 20.0, 25.0, 0.0, 0.0, None, None, None, None, 2.0, 'Ash-Vin'),
 (1, 'Heart up my Sleeves', 0.0, 1.0, 'Delhi', 'Detachable Sleeves', 'Clothes', '25 lakhs for 30% equity', 25, 25.0, 0, 0, 30.0, 10.0, '20-25', 1, 0.0, 0.0, 0.0, 0.0, 12.5, 15.0, 12.5, 15.0, 0.0, 0.0, None, None, None, None, 2.0, 'Anu-Vin'),
 (2, 'Tagz Foods', 2.0, 0.0, 'Bangalore', 'Healthy Potato Chips', 'Food', '70 lakhs for 2.75% equity', 70, 70.0, 0, 0, 2.75, 1.0, '35-40', 2, 70.0, 2.75, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 1.0, 'Ash'),
 (2, 'Head and Heart', 1.0, 1.0, 'Delhi', 'Brain Development Course', 'Health', 'No Deal', 0, 50.0, 0, 0, 0.0, 5.0, '50-55', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 0.0, None),
 (2, 'Agro tourism', 1.0, 1.0, 'Baramati', 'Tourism', 'Tourism', 'No Deal', 0, 50.0, 0, 0, 0.0, 5.0, '50-55', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 0.0, None),
 (3, 'Qzense Labs', 0.0, 2.0, 'Delhi', 'Food Freshness Detector', 'Food', 'No Deal', 0, 100.0, 0, 0, 0.0, 0.25, '30-35', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 0.0, None),
 (3, 'Peeschute', 1.0, 0.0, 'Jalna', 'Disposable Urine Bag', 'Health', '75 lakhs for 6% equity', 75, 75.0, 0, 0, 6.0, 4.0, '30-35', 1, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 75.0, 6.0, None, None, None, None, 1.0, 'Ama'),
 (3, 'NOCD', 2.0, 0.0, 'Bangalore', 'Energy Drink', 'Food', '20 lakhs for 15% equity and 30 lakhs debt', 20, 50.0, 30, 0, 15.0, 2.0, '30-35', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 50.0, 15.0, 0.0, 0.0, None, None, None, None, 1.0, 'Vin'),
 (4, 'CosIQ', 1.0, 1.0, 'Delhi', 'Intelligent Skincare', 'Health', '50 lakhs for 25% equity', 50, 50.0, 0, 0, 25.0, 7.5, '25-30', 2, 0.0, 0.0, 0.0, 0.0, 25.0, 12.5, 25.0, 12.5, 0.0, 0.0, None, None, None, None, 2.0, 'Anu-Vin'),
 (4, 'JhaJi Achaar', 0.0, 2.0, 'Darbhanga', 'Pickle', 'Food', 'No Deal', 0, 50.0, 0, 0, 0.0, 10.0, '45-50', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 0.0, None),
 (4, 'Bummer', 1.0, 0.0, 'Ahemdabad', 'Underwear', 'Clothes', '75 lakhs for 7.5% equity', 75, 75.0, 0, 0, 7.5, 4.0, '20-25', 1, 0.0, 0.0, 37.5, 3.75, 0.0, 0.0, 0.0, 0.0, 37.5, 3.75, None, None, None, None, 2.0, 'Ama-Nam'),
 (5, 'Revamp Moto', 3.0, 0.0, 'Nashik', 'E-Bike', 'Automobile', '1 crore for 1.5% equity', 100, 100.0, 0, 0, 1.5, 1.0, '25-30', 3, 0.0, 0.0, 0.0, 0.0, 50.0, 0.75, 0.0, 0.0, 50.0, 0.75, None, None, None, None, 2.0, 'Ama-Anu'),
 (5, 'Hungry Heads', 2.0, 0.0, 'Mumbai', 'Restaurant serving 80 types of Maggi', 'Food', 'No Deal', 0, 50.0, 0, 0, 0.0, 5.0, '35-40', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 0.0, None),
 (5, 'Shrawani Engineers', 2.0, 0.0, 'Pune', 'Belly Button Shaper', 'Lifestyle', 'No Deal', 0, 10.0, 0, 0, 0.0, 20.0, '35-40', 2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 0.0, None),
 (6, 'Skippi Pops', 1.0, 1.0, 'Hyderabad', 'Ice-Pops', 'Food', '1 crore for 15% equity', 100, 45.0, 0, 0, 15.0, 5.0, '35-40', 2, 20.0, 3.0, 20.0, 3.0, 20.0, 3.0, 20.0, 3.0, 20.0, 3.0, None, None, None, None, 5.0, 'Ama-Anu-Ash-Nam-Vin'),
 (6, 'Menstrupedia', 1.0, 1.0, 'Ahemdabad', 'Menstrual Awareness Comic', 'Health', '50 lakhs for 20% equity', 50, 50.0, 0, 0, 20.0, 10.0, '35-40', 2, 0.0, 0.0, 50.0, 20.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, None, None, None, None, 1.0, 'Nam'),
 (6, 'Hecolll', 0.0, 1.0, 'Hyderabad', 'Pollution Resistant Fabric', 'Clothes', 'No Deal', 0, 100.0, 0, 0, 0.0, 1.0, '35-40', 1, 0.0, 0.0, 0.0,

### Total number of Episodes

In [5]:
%sql select max(epno) from data

 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(30,)]

### Unique brands in the show

In [7]:
%sql select count(distinct brand) from data

 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(98,)]

### Pitches that got Funding

In [10]:
%sql SELECT SUM(a.converted_not_converted) AS Pitches_funded, COUNT(*) \
FROM (SELECT amountinvestedlakhs, CASE WHEN amountinvestedlakhs > 0 THEN 1 ELSE 0 END AS converted_not_converted \
      FROM data) a;


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(57, 98)]

### No. of male participants

In [11]:
%sql select sum(male) from data


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(139.0,)]

### No. of female participants

In [12]:
%sql select sum(female) from data;


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(59.0,)]

### Total amount invested by all the sharks in the show in the season 1

In [13]:
%sql select sum(amountinvestedlakhs) from data;


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(3422,)]

### Average equity taken by the sharks

In [15]:
%sql select avg(a.equitytakenp) from \
(select * from data where equitytakenp>0) a

 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(16.58035087719298,)]

### Highesh investment made by any of the shark

In [16]:
%sql select max(amountinvestedlakhs) as max_amount from data;


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(150,)]

### Highest Equity taken by any of the shark

In [17]:
%sql select max(equitytakenp) from data


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(75.0,)]

### Startups having at least one women

In [19]:
%sql SELECT SUM(a.female_count) AS total_female_count \
FROM ( \
    SELECT female, COUNT(*) AS female_count \
    FROM data \
    WHERE female > 0 \
    GROUP BY female \
) AS a;


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(52,)]

### Average teammembers

In [21]:
%sql select avg(teammembers) as Average_teammembers from data


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(2.082474226804124,)]

### Average amount invested per deal in the show

In [23]:
%sql select avg(a.amountinvestedlakhs) amount_invested_per_deal from \
(select * from data where deal!='No Deal') a

 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[(60.03508771929825,)]

### Average age group of contestants

In [24]:
%sql select avgage,count(avgage) cnt from data group by avgage order by cnt desc


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[('30-35', 32),
 ('35-40', 21),
 ('25-30', 17),
 ('20-25', 12),
 ('40-45', 6),
 ('45-50', 5),
 ('50-55', 3),
 ('41-45', 1),
 (None, 0)]

### Locations of the startups that pitched in the show

In [25]:
%sql select location,count(location) cnt from data group by location order by cnt desc


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[('Delhi', 14),
 ('Mumbai', 13),
 ('Pune', 11),
 ('Bangalore', 10),
 ('Kolkata', 6),
 ('Ahemdabad', 5),
 ('Hyderabad', 4),
 ('Gurgaon', 3),
 ('Gujarat', 3),
 ('Noida', 2),
 ('Kerela', 2),
 ('Jammu', 2),
 ('Jaipur', 2),
 ('Valsad', 1),
 ('Unknown', 1),
 ('Surat', 1),
 ('Secundarabad', 1),
 ('Panipat', 1),
 ('Nashik', 1),
 ('Nagpur', 1),
 ('Mathura', 1),
 ('Malegaon', 1),
 ('Ludhiana', 1),
 ('Lucknow', 1),
 ('Kaithal', 1),
 ('Jalna', 1),
 ('Indore', 1),
 ('Goa', 1),
 ('Dehradun', 1),
 ('Darbhanga', 1),
 ('Chennai', 1),
 ('Bombay', 1),
 ('Baramati', 1),
 (None, 0)]

### Different sectors from which the startups are

In [26]:
%sql select sector,count(sector) cnt from data group by sector order by cnt desc


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[('Food', 31),
 ('Lifestyle', 29),
 ('Health', 12),
 ('Technology', 9),
 ('Automobile', 6),
 ('Clothes', 4),
 ('food', 1),
 ('Tourism', 1),
 ('Safety', 1),
 ('Finance', 1),
 ('Entertainment', 1),
 ('Education', 1),
 (None, 0)]

### Partner Deals of the sharks

In [27]:
%sql select partners,count(partners) cnt from data  where partners!='NA' group by partners order by cnt desc


 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[('Pey', 7),
 ('Ama-Nam', 5),
 ('Nam', 3),
 ('Ash', 3),
 ('Anu-Vin', 3),
 ('Ama', 3),
 ('Gha-Pey-Vin', 2),
 ('Anu-Pey', 2),
 ('Anu', 2),
 ('Ama-Ash-Pey', 2),
 ('Ama-Ash', 2),
 ('Ama-Anu-Pey', 2),
 ('Ama-Anu-Ash-Nam-Pey', 2),
 ('Ama-Anu', 2),
 ('Vin', 1),
 ('Nam-Vin', 1),
 ('Gha-Nam-Vin', 1),
 ('Ash-Vin', 1),
 ('Ash-Nam-Pey', 1),
 ('Anu-Nam-Pey', 1),
 ('Anu-Gha-Nam-Pey-Vin', 1),
 ('Anu-Gha', 1),
 ('Anu-Ash-Pey', 1),
 ('Anu-Ash', 1),
 ('Ama-Pey', 1),
 ('Ama-Nam-Pey', 1),
 ('Ama-Ash-Vin', 1),
 ('Ama-Ash-Nam-Pey', 1),
 ('Ama-Anu-Nam-Pey', 1),
 ('Ama-Anu-Nam', 1),
 ('Ama-Anu-Ash-Pey', 1),
 ('Ama-Anu-Ash-Nam-Vin', 1)]

### Table of my 4 favorite sharks and their investments information

In [31]:
import sqlite3

conn = sqlite3.connect('C:\\Users\\mahaj\\Downloads\\data.db')
cursor = conn.cursor()

query = """
SELECT m.keyy, m.total_deals_present, m.total_deals, n.total_amount_invested, n.avg_equity_taken
FROM (
    SELECT a.keyy, a.total_Deals_present, b.total_Deals
    FROM (
        SELECT 'Ashnner' AS keyy, COUNT(ashneeramountinvested) AS total_Deals_present
        FROM data
        WHERE ashneeramountinvested != 'NA'
    ) a
    INNER JOIN (
        SELECT 'Ashnner' AS keyy, COUNT(ashneeramountinvested) AS total_Deals
        FROM data
        WHERE ashneeramountinvested != 'NA' AND ashneeramountinvested != '0'
    ) b ON a.keyy = b.keyy
) m
INNER JOIN (
    SELECT 'Ashnner' AS keyy, SUM(c.ashneeramountinvested) AS total_amount_invested, AVG(c.ashneerequitytakenp) AS avg_equity_taken
    FROM (
        SELECT * FROM data WHERE ashneerequitytakenp != 'NA' AND ashneerequitytakenp != '0'
    ) c
) n ON m.keyy = n.keyy

UNION all

SELECT m.keyy, m.total_deals_present, m.total_deals, n.total_amount_invested, n.avg_equity_taken
FROM (
    SELECT a.keyy, a.total_Deals_present, b.total_Deals
    FROM (
        SELECT 'Peyush' AS keyy, COUNT(peyushamountinvested) AS total_Deals_present
        FROM data
        WHERE peyushamountinvested != 'NA'
    ) a
    INNER JOIN (
        SELECT 'Peyush' AS keyy, COUNT(peyushamountinvested) AS total_Deals
        FROM data
        WHERE peyushamountinvested != 'NA' AND peyushamountinvested != '0'
    ) b ON a.keyy = b.keyy
) m
INNER JOIN (
    SELECT 'Peyush' AS keyy, SUM(c.peyushamountinvested) AS total_amount_invested, AVG(c.peyushequitytakenp) AS avg_equity_taken
    FROM (
        SELECT * FROM data WHERE peyushequitytakenp != 'NA' AND peyushequitytakenp != '0'
    ) c
) n ON m.keyy = n.keyy

union all

SELECT m.keyy, m.total_deals_present, m.total_deals, n.total_amount_invested, n.avg_equity_taken
FROM (
    SELECT a.keyy, a.total_Deals_present, b.total_Deals
    FROM (
        SELECT 'Aman' AS keyy, COUNT(amanamountinvested) AS total_Deals_present
        FROM data
        WHERE amanamountinvested != 'NA'
    ) a
    INNER JOIN (
        SELECT 'Aman' AS keyy, COUNT(amanamountinvested) AS total_Deals
        FROM data
        WHERE amanamountinvested != 'NA' AND amanamountinvested != '0'
    ) b ON a.keyy = b.keyy
) m
INNER JOIN (
    SELECT 'Aman' AS keyy, SUM(c.amanamountinvested) AS total_amount_invested, AVG(c.amanequitytakenp) AS avg_equity_taken
    FROM (
        SELECT * FROM data WHERE amanequitytakenp != 'NA' AND amanequitytakenp != '0'
    ) c
) n ON m.keyy = n.keyy

union ALL

SELECT m.keyy, m.total_deals_present, m.total_deals, n.total_amount_invested, n.avg_equity_taken
FROM (
    SELECT a.keyy, a.total_Deals_present, b.total_Deals
    FROM (
        SELECT 'Anupam' AS keyy, COUNT(anupamamountinvested) AS total_Deals_present
        FROM data
        WHERE anupamamountinvested != 'NA'
    ) a
    INNER JOIN (
        SELECT 'Anupam' AS keyy, COUNT(anupamamountinvested) AS total_Deals
        FROM data
        WHERE anupamamountinvested != 'NA' AND anupamamountinvested != '0'
    ) b ON a.keyy = b.keyy
) m
INNER JOIN (
    SELECT 'Anupam' AS keyy, SUM(c.anupamamountinvested) AS total_amount_invested, AVG(c.anupamequitytakenp) AS avg_equity_taken
    FROM (
        SELECT * FROM data WHERE anupamequitytakenp != 'NA' AND anupamequitytakenp != '0'
    ) c
) n ON m.keyy = n.keyy;

"""

cursor.execute(query)

results = cursor.fetchall()
print(results)

conn.commit()
conn.close()


[('Ashnner', 79, 17, 560.83, 4.140196078411765), ('Peyush', 71, 24, 784.0300000000001, 13.723333333333334), ('Aman', 78, 24, 799.9300000000001, 6.095), ('Anupam', 96, 20, 518.8, 7.638)]


### As we are using Jupyter notebook, it doesn't support table format properly which SQL does. So just to show this information properly i will be using Pandas Dataframe.

In [36]:
data = [('Ashnner', 79, 17, 560.83, 4.140196078411765), ('Peyush', 71, 24, 784.0300000000001, 13.723333333333334), ('Aman', 78, 24, 799.9300000000001, 6.095), ('Anupam', 96, 20, 518.8, 7.638)]

In [38]:
import pandas as pd
columns = ['keyy', 'total_deals_present', 'total_deals', 'total_amount_invested', 'avg_equity_taken']
df = pd.DataFrame(data, columns=columns)

df

keyy  total_deals_present  total_deals  total_amount_invested  \
0  Ashnner                   79           17                 560.83   
1   Peyush                   71           24                 784.03   
2     Aman                   78           24                 799.93   
3   Anupam                   96           20                 518.80   

   avg_equity_taken  
0          4.140196  
1         13.723333  
2          6.095000  
3          7.638000

#### Exact same table can be seen if you are using any of the SQL IDE

### Startups in which the highest amount has been invested in each domain/sector

In [33]:
%sql SELECT c.*\
FROM (\
    SELECT brand,sector, amountinvestedlakhs, RANK() OVER (PARTITION BY sector ORDER BY amountinvestedlakhs DESC) AS rnk\
    FROM data\
) c\
WHERE c.rnk = 1;

 * sqlite:///C:/Users/mahaj/Downloads/data.db
Done.


[('Watt Technovations', None, None, 1),
 ('Revamp Moto', 'Automobile', 100, 1),
 ('The Yarn Bazaar', 'Clothes', 100, 1),
 ('Scholify', 'Education', 0, 1),
 ('Kabaddi Adda', 'Entertainment', 80, 1),
 ('Insurance Samadhan', 'Finance', 100, 1),
 ('Skippi Pops', 'Food', 100, 1),
 ('IN A CAN', 'Food', 100, 1),
 ('Get a Whey', 'Food', 100, 1),
 ('Humpy A2', 'Food', 100, 1),
 ('Annie', 'Health', 105, 1),
 ('Raising Superstars', 'Lifestyle', 100, 1),
 ('Hammer Lifestyle', 'Lifestyle', 100, 1),
 ('Gold Safe Solutions Ind.', 'Safety', 50, 1),
 ('Aas Vidyalaya', 'Technology', 150, 1),
 ('Agro tourism', 'Tourism', 0, 1),
 ('Theka Coffee', 'food', 0, 1)]

### Insights Observed: 

- <p>Out of 98 pitches 57 pitches got funded on the show

- <p>Average Equity taken by sharks is around 15%

- <p>Highest investement made on the show is 1.50 Cr Rupees

- <p>Highest equity taken on the show is 75%

- <p>Most of the contestants are from the age-group of 30-35

- <p>Most of the startups are from Delhi, Mumbai, Pune and Bangalore

- <p>Aman, Namita and Anupam and Vineeta are the 2 pair of sharks which partnered on most of the deals

- <p>Technology, Automobile, Food and Entertainment are the sectors where sharks made heavy investments